packages/devtools_app/lib/src/screens/performance/enhance_tracing.dart

```diff
@@ -0,0 +1,379 @@
+        extensions.profileRenderObjectLayouts,
+        extensions.profileRenderObjectPaints,
+      ],
+      customExtensionUi: {
+        extensions.profileWidgetBuilds.extension:
+            const TrackWidgetBuildsSetting(),
+        extensions.profileUserWidgetBuilds.extension: const SizedBox(),
+      },
+      overlayDescription: RichText(
+        text: TextSpan(
+          text: 'These options can be used to add more detail to the '
+              'timeline, but be aware that ',
+          style: textStyle,
+          children: [
+            TextSpan(
+              text: 'frame times may be negatively affected',
+              style:
+                  textStyle.copyWith(color: theme.colorScheme.errorTextColor),
+            ),
+            TextSpan(
+              text: '.\n\n',
+              style: textStyle,
+            ),
+            TextSpan(
+              text: 'When toggling on/off a tracing option, you will need '
+                  'to reproduce activity in your app to see the enhanced '
+                  'tracing in the timeline.',
+              style: textStyle,
+            )
+          ],
+        ),
+      ),
+    );
+  }
+}
+
+enum TrackWidgetBuildsScope {
+  all,
+  userCreated,
+}
+
+extension TrackWidgetBuildsScopeExtension on TrackWidgetBuildsScope {
+  String get radioDisplay {
+    switch (this) {
+      case TrackWidgetBuildsScope.all:
+        return 'within all code';
+      case TrackWidgetBuildsScope.userCreated:
+        return 'within your code';
+    }
+  }
+
+  /// Returns the opposite [TrackWidgetBuildsScope] from [this].
+  TrackWidgetBuildsScope get opposite {
+    switch (this) {
+      case TrackWidgetBuildsScope.all:
+        return TrackWidgetBuildsScope.userCreated;
+      case TrackWidgetBuildsScope.userCreated:
+        return TrackWidgetBuildsScope.all;
+    }
+  }
+
+  /// Returns the service extension for this [TrackWidgetBuildsScope].
+  extensions.ToggleableServiceExtensionDescription<bool> get extensionForScope {
+    switch (this) {
+      case TrackWidgetBuildsScope.all:
+        return extensions.profileWidgetBuilds;
+      case TrackWidgetBuildsScope.userCreated:
+        return extensions.profileUserWidgetBuilds;
+    }
+  }
+}
+
+class TrackWidgetBuildsSetting extends StatefulWidget {
+  const TrackWidgetBuildsSetting({Key? key}) : super(key: key);
+
+  @override
+  State<TrackWidgetBuildsSetting> createState() =>
+      _TrackWidgetBuildsSettingState();
+}
+
+class _TrackWidgetBuildsSettingState extends State<TrackWidgetBuildsSetting>
+    with AutoDisposeMixin {
+  static const _scopeSelectorPadding = 32.0;
+
+  /// Service extensions for tracking widget builds.
+  ///
+  /// There is an extension for each value of [TrackWidgetBuildsScope], and the
+  /// extensions in this list are indexed according to their enum value index.
+  final _trackWidgetBuildsExtensions = [
+    extensions.profileWidgetBuilds,
+    extensions.profileUserWidgetBuilds,
+  ];
```

Comment:
Using a list can save space since with a map, we are keeping a set of keys in memory that we do not actually need. But for this case, since it is small, I have replaced with a map for improved readability.